pennylane/qnn.py

```diff
@@ -0,0 +1,205 @@
+    satisfies the following conditions:
+
+    - Contain an ``inputs`` argument for inputting data. All other arguments are treated as
+      weights within the QNode.
+    - All arguments must accept an array or tensor, e.g., arguments should not use nested lists
+      of different lengths.
+    - All arguments, except ``inputs``, must have no default value.
+    - The ``inputs`` argument is permitted to have a default value provided the gradient with
+      respect to ``inputs`` is not required.
+    - There cannot be a variable number of positional or keyword arguments, e.g., no ``*args`` or
+      ``**kwargs`` present in the signature.
+
+    The QNode weights are initialized within the :class:`~.KerasLayer`. Upon instantiation,
+    a ``weight_shapes`` dictionary must hence be passed which describes the shapes of all
+    weights in the QNode. The optional ``weight_specs`` argument allows for a more fine-grained
+    specification of the QNode weights, such as the method of initialization and any
+    regularization or constraints. If not specified, weights will be added using the Keras
+    default initialization and without any regularization
+    or constraints.
+
+    **Example usage:**
+
+    .. code-block:: python
+
+        n_qubits = 2
+        dev = qml.dev("default.qubit", wires=n_qubits)
+
+        @qml.qnode(dev, interface="tf")
+        def circuit(inputs, weights):
+            qml.templates.AngleEmbedding(inputs, wires=list(range(n_qubits)))
+            qml.templates.StronglyEntanglingLayers(weights, wires=list(range(n_qubits)))
+            return qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliZ(1))
+
+        weight_shapes = {"weights": (3, n_qubits, 3)}
+        weight_specs = {"weights": {"initializer": "random_uniform"}}
+
+        qnode = qml.qnn.KerasLayer(circuit, weight_shapes, 2, weight_specs)
+
+    The resulting ``qnode`` can be treated as a Keras layer and can be combined with other layers
+    using the `Sequential <https://www.tensorflow.org/api_docs/python/tf/keras/Sequential>`__ or
+    `Model <https://www.tensorflow.org/api_docs/python/tf/keras/Model>`__ APIs.
+
+    Args:
+        qnode (qml.QNode): the PennyLane QNode to be converted into a Keras layer
+        weight_shapes (dict[str, tuple]): a dictionary mapping from all weights used in the QNode to
+            their corresponding sizes
+        output_dim (int): the dimension of data output from the QNode
+        weight_specs (dict[str, dict]): an optional dictionary for users to provide additional
+            specifications for weights used in the QNode, such as the method of parameter
+            initialization. This specification is provided as a dictionary with keys given by the
+            arguments of the `add_weight()
+            <https://www.tensorflow.org/api_docs/python/tf/keras/layers/Layer#add_weight>`__
+            (e.g., ``initializer``) method and values being the corresponding specification.
```

Comment:
just confirming, because it reads awkwardly

Is it necessary for the bracket to appear before "method", or can it also appear right after? or does it need to appear at all>